packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -64,6 +70,28 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
     throw new Error(`The stack ${options.stack.displayName} does not have an environment`);
   }
 
+  const cfn = await options.sdk.cloudFormation(options.stack.environment.account, options.stack.environment.region, Mode.ForWriting);
+  const deployName = options.deployName || options.stack.stackName;
+
+  if (!options.force) {
+    debug(`checking if we can skip this stack based on the currently deployed template (use --force to override)`);
+    const deployed = await getDeployedTemplate(cfn, deployName);
+    if (deployed && JSON.stringify(options.stack.template) === JSON.stringify(deployed.template)) {
+      debug(`${deployName}: no change in template, skipping (use --force to override)`);
```

Comment:
i'm wondering if this log is valuable just as a warning. The prescriptive action to remedy is something we'd want customers to see, and it might be surprising if nothing happens without warning.

the down side is it may create clutter if there are a lot of no-ops but maybe we can consider turning it into a debug log at a later time

Reply:
Since this will (hopefully) will just work for 99% of the users, I don't think this is something that we need all users to see all the time. If something is broken, they can always use `--verbose` and see this.

Reply:
seems reasonable, it's a 2-way door in any case. we can always change if it causes any confusion